pennylane/templates/layers/custom_entangler.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``BasicEntanglerLayers`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, RX
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    check_shape,
+    check_no_variable,
+    check_number_of_layers,
+    get_shape,
+    check_shapes,
+)
+from pennylane.wires import Wires
+
+
+@template
+def CustomEntanglerLayers(
+    rotation_weights, wires, rotation=None, coupling=None, coupling_weights=None, pattern=None
```

Comment:
Before merging, it's good to think about the ordering of the keyword arguments.

Some things to keep in mind:
- signatures should mirror other similar functions as much as possible
- the arguments that the user would likely use most should be the earliest in the signature (since python lets you specify keyword args positionally sometimes)

Reply:
I'm thinking that it should probably be `pattern`, `rotation`, `coupling`, `coupling_weights`, what did you have in mind?